### PR TITLE
Drop the dataset batch endpoint

### PR DESCRIPTION
**Pull request by mcwhittemore**
As we look to move the dataset api past beta we have decided that that batch api is not something we plan to support for the long term.

This PR drops support of it fully rather than converting the batch functionality to use the single feature put and delete endpoints.

@scothis for the review

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 1.0.0.beta4
```
Comment (scothis) [inline CHANGELOG.md]: `-beta4`

File: lib/services/datasets.js [hunk 2]
```diff
@@ -38,14 +38,21 @@ var Datasets = module.exports = makeService('MapboxDatasets');
...
     path: constants.API_DATASET_DATASETS,
     params: {
+      limit: opts.limit,
+      start: opts.start,
+      fresh: opts.fresh,
       owner: this.owner
     },
     callback: callback
-  }).entity();
+  });
```
Comment (scothis) [inline lib/services/datasets.js]: Curious if @tmcw has any concern about dropping `.entity()`.

**Comment by mcwhittemore** [comment]
@scothis do you think the promise response should match the callback response object?

**Comment by scothis** [comment]
Yes, the callback and promise should hold the same value.

**Comment by mcwhittemore** [comment]
@scothis - ok, I think this is ready now.

File: lib/standard_response.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+var interceptor = require('rest/interceptor');
+
+var standardResponse = interceptor({
+  response: transform,
+});
+
+function transform(response) {
+  return {
+    url: response.url,
+    status: response.status.code,
```
Comment (scothis) [inline lib/standard_response.js]: It's possible some errors may not have a `status` object.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+## 1.0.0
```
Comment (scothis) [inline CHANGELOG.md]: Are we ready for `1.0`?

Comment (mcwhittemore) [inline CHANGELOG.md]: IDK? We've been on `beta` a few times now. In an earlier commit, I called this `beta4`. I can go back to that if you like.

What is our goal with 1.0?

**Pull request merged**